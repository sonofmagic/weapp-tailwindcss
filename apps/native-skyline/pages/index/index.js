import { cityData } from './data'
import { add } from 'lodash'
Component({
  data: {
    list: [],
  },
  lifetimes: {
    attached() {
      const cities = cityData
      // 按拼音排序
      cities.sort((c1, c2) => {
        const pinyin1 = c1.pinyin.join('')
        const pinyin2 = c2.pinyin.join('')
        return pinyin1.localeCompare(pinyin2)
      })
      // 添加首字母
      const map = new Map()
      for (const city of cities) {
        const alpha = city.pinyin[0].charAt(0).toUpperCase()
        if (!map.has(alpha)) map.set(alpha, [])
        map.get(alpha).push({ name: city.fullname })
      }

      const keys = []
      for (const key of map.keys()) {
        keys.push(key)
      }
      keys.sort()

      const list = []
      for (const key of keys) {
        list.push({
          alpha: key,
          subItems: map.get(key)
        })
      }

      console.log('address-book list:', list)
      this.setData({ list })
      console.log(add(1,3));
    },
  },
})
