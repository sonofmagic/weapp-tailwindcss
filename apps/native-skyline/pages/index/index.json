{
  "usingComponents": {
    "navigation-bar": "../../components/navigation-bar",
    "address-book": "../../components/address-book"
  },
  "disableScroll": true,
  "navigationStyle": "custom"
}