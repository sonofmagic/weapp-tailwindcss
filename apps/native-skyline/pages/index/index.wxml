<navigation-bar title="通讯录" back="{{true}}"></navigation-bar>
<view class="w-screen grow overflow-hidden">
  <address-book list="{{list}}">
    <view class="text-black/90">
      <view class="p-10">
        <view class="page__title">Address Book</view>
        <view class="page__desc">类通讯录列表</view>
        <view class="i-mdi-home bg-yellow-600 w-4 h-4"></view>
      </view>
      <view class="pb-10">
      </view>
    </view>
  </address-book>
</view>
