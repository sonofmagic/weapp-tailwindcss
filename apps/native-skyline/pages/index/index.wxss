page {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  height: 100vh;
  background-color: #f7f7f7;
}

.page__title {
  @apply text-center text-xl;
}

.page__desc {
  @apply text-center text-sm;
  margin-top: 5px;
  color: rgba(0, 0, 0, 0.5);
}

.cell {
  height: 50px;
  justify-content: center;
  align-items: center;
}
