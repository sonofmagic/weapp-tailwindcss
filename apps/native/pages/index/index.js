// index.js
// 获取应用实例
const app = getApp()

Page({
  data: {
    motto: 'Hello World',
    userInfo: {},
    hasUserInfo: false,
    canIUse: wx.canIUse('button.open-type.getUserInfo'),
    imgUrls: null,
    indicatorDots: true,
    autoplay: true,
    interval: 5000,
    duration: 1000,
    list: [],
    page: 1
  },
  // 事件处理函数
  bindViewTap: function () {
    wx.navigateTo({
      url: '../logs/logs'
    })
  },
  addCart(data) {
    const item = data.currentTarget.dataset.item
  },
  imgsc: function (e) {
    wx.chooseImage({
      count: 1, // 默认9
      sizeType: ['original', 'compressed'], // 可以指定是原图还是压缩图，默认二者都有
      sourceType: ['album', 'camera'], // 可以指定来源是相册还是相机，默认二者都有
      success: function (res) {
        // 返回选定照片的本地文件路径列表，tempFilePath可以作为img标签的src属性显示图片
        const tempFilePaths = res.tempFilePaths
      }
    })
  },
  lower: function (e) {
    console.log(e)
    this.getList()
  },
  getList: function () {},
  onLoad: function () {
    const app = getApp()
    this.getList()

    if (app.globalData.userInfo) {
      this.setData({
        userInfo: app.globalData.userInfo,
        hasUserInfo: true
      })
    } else if (this.data.canIUse) {
      // 由于 getUserInfo 是网络请求，可能会在 Page.onLoad 之后才返回
      // 所以此处加入 callback 以防止这种情况
      app.userInfoReadyCallback = (res) => {
        this.setData({
          userInfo: res.userInfo,
          hasUserInfo: true
        })
      }
    } else {
      // 在没有 open-type=getUserInfo 版本的兼容处理
      wx.getUserInfo({
        success: (res) => {
          app.globalData.userInfo = res.userInfo
          this.setData({
            userInfo: res.userInfo,
            hasUserInfo: true
          })
        }
      })
    }
  },
  changeIndicatorDots: function (e) {
    this.setData({
      indicatorDots: !this.data.indicatorDots
    })
  },
  changeAutoplay: function (e) {
    this.setData({
      autoplay: !this.data.autoplay
    })
  },
  intervalChange: function (e) {
    this.setData({
      interval: e.detail.value
    })
  },
  durationChange: function (e) {
    this.setData({
      duration: e.detail.value
    })
  },
  getUserInfo: function (e) {
    app.globalData.userInfo = e.detail.userInfo
    this.setData({
      userInfo: e.detail.userInfo,
      hasUserInfo: true
    })
  }
})
