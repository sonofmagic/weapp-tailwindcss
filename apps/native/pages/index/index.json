{
  "navigationBarTitleText": "首页",
  "enablePullDownRefresh":true,
  "backgroundTextStyle":"dark"
}