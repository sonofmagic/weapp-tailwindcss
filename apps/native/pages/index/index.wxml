<!-- index.wxml -->
<scroll-view class="scrollarea" scroll-y type="list">
  <view class="container">
    <view class="userinfo">
      <block wx:if="{{canIUseNicknameComp && !hasUserInfo}}">
        <button open-type="chooseAvatar" bind:chooseavatar="onChooseAvatar">
          <image class="w-[100px] h-[100px]" src="{{userInfo.avatarUrl}}"></image>
        </button>
        <view class="nickname-wrapper">
          <text class="nickname-label">昵称</text>
          <input type="nickname" class="nickname-input" placeholder="请输入昵称" bind:change="onInputChange" />
        </view>
        <view class="text-[#432432]">{{ num }}</view>
      </block>
      <block wx:elif="{{!hasUserInfo}}">
        <button wx:if="{{canIUseGetUserProfile}}" bindtap="getUserProfile">获取头像昵称</button>
        <view wx:else>请使用2.10.4及以上版本基础库</view>
      </block>
      <block wx:else>
        <image bindtap="bindViewTap" class="userinfo-avatar" src="{{userInfo.avatarUrl}}" mode="cover"></image>
        <text>{{userInfo.nickName}}</text>
      </block>
    </view>
    <view class="mt-[40px]">
      <text class="{{ motto }}">{{motto}}</text>
    </view>
  </view>
</scroll-view>