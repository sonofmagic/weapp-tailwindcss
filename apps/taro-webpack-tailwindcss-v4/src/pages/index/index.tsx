import { Text, View } from '@tarojs/components'
import { useLoad } from '@tarojs/taro'
import './index.css'

export default function Index() {
  useLoad(() => {
    console.log('Page loaded.')
  })

  return (
    <>
      <View className="bg-[#2e2bcc] text-[100rpx] text-white">
        <Text>Hello world!</Text>
      </View>
      <View className="bg-purple-800 text-pink-200">
        11
      </View>
      <View className="h-14 bg-gradient-to-r from-cyan-500 to-blue-500"></View>
      <View className="bg-purple-800 text-pink-200 invert">
        invert
      </View>
      <View className="bg-purple-800 text-pink-200 backdrop-blur-2xl">
        backdrop-blur-2xl
      </View>
      <View className="skew-x-0 bg-purple-800 text-pink-200">
        skew-x-0
      </View>
      <View className="flex space-x-1 space-y-1.5">
        <View className="bg-purple-800 text-pink-200">
          Tailwind
        </View>
        <View className="bg-purple-800 text-pink-200">
          Tailwind
        </View>
        <View className="bg-purple-800 text-pink-200">
          Tailwind
        </View>
        <Text className="bg-purple-800 text-pink-200">
          Tailwind
        </Text>
      </View>
      <View className="flex flex-col space-y-1.5">
        <View className="bg-purple-800 text-pink-200">
          Tailwind CSS
        </View>
        <View className="bg-purple-800 text-pink-200">
          Tailwind CSS
        </View>
        <View className="bg-purple-800 text-pink-200">
          Tailwind CSS
        </View>
        <Text className="bg-purple-800 text-pink-200">
          Tailwind CSS
        </Text>
      </View>
    </>

  )
}
