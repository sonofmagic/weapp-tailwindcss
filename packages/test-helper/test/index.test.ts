import fs from 'fs-extra'
import path from 'pathe'
import { generateCss3, generateCss4 } from '@/index'

const targetWxmlPath = path.resolve(import.meta.dirname, './fixtures/v4/index.wxml')

const targetWxml = await fs.readFile(targetWxmlPath, 'utf8')

describe('tailwindcss', () => {
  describe('3', () => {
    it('should ', async () => {
      expect((await generateCss3(targetWxml)).css).toMatchSnapshot()
    })

    it('should 2', async () => {
      expect((await generateCss3(targetWxml, {
        css: `@tailwind base;@tailwind components;@tailwind utilities;`,
      })).css).toMatchSnapshot()
    })
  })

  describe('4', () => {
    it('should ', async () => {
      expect((await generateCss4(
        path.dirname(targetWxmlPath),
      )).css).toMatchSnapshot()
    })
  })
})
