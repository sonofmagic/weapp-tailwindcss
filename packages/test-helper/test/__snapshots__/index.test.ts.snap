// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tailwindcss > 3 > should  1`] = `
".mt-\\[13\\.43px\\] {
    margin-top: 13.43px
}
.bg-sky-300 {
    --tw-bg-opacity: 1;
    background-color: rgb(125 211 252 / var(--tw-bg-opacity, 1))
}
.text-red-100 {
    --tw-text-opacity: 1;
    color: rgb(254 226 226 / var(--tw-text-opacity, 1))
}"
`;

exports[`tailwindcss > 3 > should 2 1`] = `
"*, ::before, ::after {
  --tw-border-spacing-x: 0;
  --tw-border-spacing-y: 0;
  --tw-translate-x: 0;
  --tw-translate-y: 0;
  --tw-rotate: 0;
  --tw-skew-x: 0;
  --tw-skew-y: 0;
  --tw-scale-x: 1;
  --tw-scale-y: 1;
  --tw-pan-x:  ;
  --tw-pan-y:  ;
  --tw-pinch-zoom:  ;
  --tw-scroll-snap-strictness: proximity;
  --tw-gradient-from-position:  ;
  --tw-gradient-via-position:  ;
  --tw-gradient-to-position:  ;
  --tw-ordinal:  ;
  --tw-slashed-zero:  ;
  --tw-numeric-figure:  ;
  --tw-numeric-spacing:  ;
  --tw-numeric-fraction:  ;
  --tw-ring-inset:  ;
  --tw-ring-offset-width: 0px;
  --tw-ring-offset-color: #fff;
  --tw-ring-color: rgb(59 130 246 / 0.5);
  --tw-ring-offset-shadow: 0 0 #0000;
  --tw-ring-shadow: 0 0 #0000;
  --tw-shadow: 0 0 #0000;
  --tw-shadow-colored: 0 0 #0000;
  --tw-blur:  ;
  --tw-brightness:  ;
  --tw-contrast:  ;
  --tw-grayscale:  ;
  --tw-hue-rotate:  ;
  --tw-invert:  ;
  --tw-saturate:  ;
  --tw-sepia:  ;
  --tw-drop-shadow:  ;
  --tw-backdrop-blur:  ;
  --tw-backdrop-brightness:  ;
  --tw-backdrop-contrast:  ;
  --tw-backdrop-grayscale:  ;
  --tw-backdrop-hue-rotate:  ;
  --tw-backdrop-invert:  ;
  --tw-backdrop-opacity:  ;
  --tw-backdrop-saturate:  ;
  --tw-backdrop-sepia:  ;
  --tw-contain-size:  ;
  --tw-contain-layout:  ;
  --tw-contain-paint:  ;
  --tw-contain-style:  ;
}

::backdrop {
  --tw-border-spacing-x: 0;
  --tw-border-spacing-y: 0;
  --tw-translate-x: 0;
  --tw-translate-y: 0;
  --tw-rotate: 0;
  --tw-skew-x: 0;
  --tw-skew-y: 0;
  --tw-scale-x: 1;
  --tw-scale-y: 1;
  --tw-pan-x:  ;
  --tw-pan-y:  ;
  --tw-pinch-zoom:  ;
  --tw-scroll-snap-strictness: proximity;
  --tw-gradient-from-position:  ;
  --tw-gradient-via-position:  ;
  --tw-gradient-to-position:  ;
  --tw-ordinal:  ;
  --tw-slashed-zero:  ;
  --tw-numeric-figure:  ;
  --tw-numeric-spacing:  ;
  --tw-numeric-fraction:  ;
  --tw-ring-inset:  ;
  --tw-ring-offset-width: 0px;
  --tw-ring-offset-color: #fff;
  --tw-ring-color: rgb(59 130 246 / 0.5);
  --tw-ring-offset-shadow: 0 0 #0000;
  --tw-ring-shadow: 0 0 #0000;
  --tw-shadow: 0 0 #0000;
  --tw-shadow-colored: 0 0 #0000;
  --tw-blur:  ;
  --tw-brightness:  ;
  --tw-contrast:  ;
  --tw-grayscale:  ;
  --tw-hue-rotate:  ;
  --tw-invert:  ;
  --tw-saturate:  ;
  --tw-sepia:  ;
  --tw-drop-shadow:  ;
  --tw-backdrop-blur:  ;
  --tw-backdrop-brightness:  ;
  --tw-backdrop-contrast:  ;
  --tw-backdrop-grayscale:  ;
  --tw-backdrop-hue-rotate:  ;
  --tw-backdrop-invert:  ;
  --tw-backdrop-opacity:  ;
  --tw-backdrop-saturate:  ;
  --tw-backdrop-sepia:  ;
  --tw-contain-size:  ;
  --tw-contain-layout:  ;
  --tw-contain-paint:  ;
  --tw-contain-style:  ;
}

*,
::before,
::after {
  box-sizing: border-box;
  border-width: 0;
  border-style: solid;
  border-color: #e5e7eb;
}

::before,
::after {
  --tw-content: '';
}

html,
:host {
  line-height: 1.5;
  -webkit-text-size-adjust: 100%;
  -moz-tab-size: 4;
  tab-size: 4;
  font-family: ui-sans-serif, system-ui, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
  font-feature-settings: normal;
  font-variation-settings: normal;
  -webkit-tap-highlight-color: transparent;
}

body {
  margin: 0;
  line-height: inherit;
}

hr {
  height: 0;
  color: inherit;
  border-top-width: 1px;
}

abbr:where([title]) {
  text-decoration: underline dotted;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  font-size: inherit;
  font-weight: inherit;
}

a {
  color: inherit;
  text-decoration: inherit;
}

b,
strong {
  font-weight: bolder;
}

code,
kbd,
samp,
pre {
  font-family: ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
  font-feature-settings: normal;
  font-variation-settings: normal;
  font-size: 1em;
}

small {
  font-size: 80%;
}

sub,
sup {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
}

sub {
  bottom: -0.25em;
}

sup {
  top: -0.5em;
}

table {
  text-indent: 0;
  border-color: inherit;
  border-collapse: collapse;
}

button,
input,
optgroup,
select,
textarea {
  font-family: inherit;
  font-feature-settings: inherit;
  font-variation-settings: inherit;
  font-size: 100%;
  font-weight: inherit;
  line-height: inherit;
  letter-spacing: inherit;
  color: inherit;
  margin: 0;
  padding: 0;
}

button,
select {
  text-transform: none;
}

button,
input:where([type='button']),
input:where([type='reset']),
input:where([type='submit']) {
  -webkit-appearance: button;
  background-color: transparent;
  background-image: none;
}

:-moz-focusring {
  outline: auto;
}

:-moz-ui-invalid {
  box-shadow: none;
}

progress {
  vertical-align: baseline;
}

::-webkit-inner-spin-button,
::-webkit-outer-spin-button {
  height: auto;
}

[type='search'] {
  -webkit-appearance: textfield;
  outline-offset: -2px;
}

::-webkit-search-decoration {
  -webkit-appearance: none;
}

::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

summary {
  display: list-item;
}

blockquote,
dl,
dd,
h1,
h2,
h3,
h4,
h5,
h6,
hr,
figure,
p,
pre {
  margin: 0;
}

fieldset {
  margin: 0;
  padding: 0;
}

legend {
  padding: 0;
}

ol,
ul,
menu {
  list-style: none;
  margin: 0;
  padding: 0;
}
dialog {
  padding: 0;
}

textarea {
  resize: vertical;
}

input::placeholder,
textarea::placeholder {
  opacity: 1;
  color: #9ca3af;
}

button,
[role="button"] {
  cursor: pointer;
}
:disabled {
  cursor: default;
}

img,
svg,
video,
canvas,
audio,
iframe,
embed,
object {
  display: block;
  vertical-align: middle;
}

img,
video {
  max-width: 100%;
  height: auto;
}
[hidden]:where(:not([hidden="until-found"])) {
  display: none;
}.mt-\\[13\\.43px\\] {
  margin-top: 13.43px;
}.bg-sky-300 {
  --tw-bg-opacity: 1;
  background-color: rgb(125 211 252 / var(--tw-bg-opacity, 1));
}.text-red-100 {
  --tw-text-opacity: 1;
  color: rgb(254 226 226 / var(--tw-text-opacity, 1));
}"
`;

exports[`tailwindcss > 4 > should  1`] = `
":root, :host {
  --color-red-100: oklch(93.6% .032 17.717);
  --color-sky-300: oklch(82.8% .111 230.318);
}
*,:after,:before,::backdrop {
  box-sizing: border-box;
  border: 0 solid;
  margin: 0;
  padding: 0;
}
.mt-\\[13\\.43px\\] {
  margin-top: 13.43px;
}
.bg-sky-300 {
  background-color: var(--color-sky-300);
}
.text-red-100 {
  color: var(--color-red-100);
}"
`;
